src/App.js

```diff
@@ -1,31 +1,89 @@
 import React from 'react';
+import DemoLayout from './DemoLayout.json'
 import RGL, { WidthProvider } from 'react-grid-layout';
-import sounds from './sounds.js'
+import SoundButton from './SoundButton.js';
 import ToggleButton from 'react-bootstrap/ToggleButton';
 import ToggleButtonGroup from 'react-bootstrap/ToggleButtonGroup';
 import { Line } from 'rc-progress';
+import _ from "lodash";
 import './App.css';
 import '../node_modules/react-grid-layout/css/styles.css'
 import '../node_modules/react-resizable/css/styles.css'
 
 const GridLayout = WidthProvider(RGL);
 
 class Container extends React.Component {
+  static defaultProps = {
+    defaultLayout: DemoLayout
+  }
+
   constructor(props){
     super(props)
     this.state = {
       progress: 0,
       editMode: false,
-      playing: false
+      playing: false,
+      layoutRaw: this.props.defaultLayout,
+      layout: this.generateLayout(this.props.defaultLayout)
     }
+
+    this.audio = this.importBulk(require.context('./sounds', false, /\.(wav|mp3)$/))
+
     this.currentAudio = null
+    this.playing = false
     this.interval = null
 
-    this.changeEditMode = this.changeEditMode.bind(this)
-    this.playSound = this.playSound.bind(this)
+    this.handleEditModeChange = this.handleEditModeChange.bind(this)
+    this.handleSoundButtonClick = this.handleSoundButtonClick.bind(this)
+  }
+
+  importBulk(r) {
```

Comment:
Can you write some comments for this method? For example, what is `r`? What does it return?

Reply:
Bump.

Reply:
Added a comment and made it more readable.